scripts/west_commands/build.py

```diff
@@ -175,26 +183,27 @@ def do_run(self, args, remainder):
         self._setup_source_dir()
         self._sanity_check()
 
-        log.inf('source directory: {}'.format(self.source_dir), colorize=True)
-        log.inf('build directory: {}{}'.
-                format(self.build_dir,
-                       (' (created)' if self.created_build_dir
-                        else '')),
-                colorize=True)
-        if self.cmake_cache:
-            board = self.cmake_cache.get('CACHED_BOARD')
-        elif self.args.board:
-            board = self.args.board
-        else:
-            board = 'UNKNOWN'   # shouldn't happen
-        log.inf('BOARD:', board, colorize=True)
-
-        self._run_cmake(self.args.cmake_opts)
+        board, origin = self._find_board()
+        self._run_cmake(board, origin, self.args.cmake_opts)
         self._sanity_check()
         self._update_cache()
 
         self._run_build(args.target)
 
+    def _find_board(self):
+        board, origin = None, None
+        config_board = config_get('board', None)
+        if self.cmake_cache:
+            board, origin = (self.cmake_cache.get('CACHED_BOARD'),
```

Comment:
Can this ever return `None` now that I think of it?

Reply:
Not if the build directory is valid

Reply:
Was on mobile before. To amplify: if the build directory is messed up somehow, board comes back None, but `_run_cmake` complains asking for `--board`. At that point, if the user passes `--board` and the build *still* doesn't work, something is quite wrong with the build directory and they probably just need to delete it and start over.